src/main/java/Duke.java

```diff
@@ -1,10 +1,144 @@
+                    }
+                }
+            } else if (command.startsWith("mark ") || command.startsWith("unmark ")) {
+                try {
+                    // Mark task as done
+                    if (command.substring(0, command.indexOf(" ")).equalsIgnoreCase("mark")) {
+                        int index = Integer.parseInt(command.replaceAll("mark ", "")) - 1;
+                        if (taskList.get(index).isDone()) {
+                            throw new DukeException("marked");
+                        }
+                        else {
+                            taskList.get(index).mark();
+                            tasksDone += 1;
+                        }
+                    }
+                    // Unmark task
+                    else if (command.substring(0, command.indexOf(" ")).equalsIgnoreCase("unmark")) {
+                        int index = Integer.parseInt(command.replaceAll("unmark ", "")) - 1;
+                        if (taskList.get(index).isDone()) {
+                            taskList.get(index).unmark();
+                            tasksDone -= 1;
+                        }
+                        else {
+                            throw new DukeException("unmarked");
+                        }
+                    }
+                } catch (DukeException e) {
+                    e.MarkedException();
+                }
+            }
+            // Delete a task
+            else if (command.toLowerCase().startsWith("delete ")) {
+                int index = Integer.parseInt(command.replaceAll("delete ", "")) - 1;
+                Task tempTask = taskList.get(index);
+                taskList.remove(index);
+                System.out.println("Noted. I've removed this task:" + "\n" + tempTask.toString());
+            }
+            else {
+                try {
+                    Task newTask;
+
+                    // Create new ToDo task
+                    if (command.toLowerCase().startsWith("todo ")) {
+                        String description = command.substring(command.indexOf(" ") + 1);
+
+                        if (description.equals("")) {
+                            throw new DukeException("todo");
+                        }
+                        else{
+                            newTask = new ToDo(command.substring(command.indexOf(" ") + 1));
+                        }
+                    }
+                    // Create new Deadline task
+                    else if (command.toLowerCase().startsWith("deadline ")) {
+                        String description = command.substring(command.indexOf(" ") + 1, command.indexOf("/"));
+                        String by = command.substring(command.indexOf("/by") + 4);
+
+                        System.out.println(description + " - " + by);
+
+                        if (description.equals("")) {
+                            throw new DukeException("deadline");
```

Comment:
You may want to include a more specific error message when you throw Exceptions.